File: 0645-set-mismatch/0645-set-mismatch.cpp
class Solution {
public:
    vector<int> findErrorNums(vector<int>& nums) {
        unordered_map<int, bool>seen;
        vector<int>res(2);
        for (auto num: nums) {
            if (seen.find(num) != seen.end()) {
                res[0] = num;
                continue;
            }
            seen[num] = true;
        }
        
        for (int i = 1; i < nums.size() + 1; i++) {
            if (seen.find(i) == seen.end()) {
                res[1] = i;
                break;
            }
        }
        
        return res;
    }
};

File: 0645-set-mismatch/0645-set-mismatch.cs
public class Solution {
    public int[] FindErrorNums(int[] nums) {
        int n = nums.Length;
        var s = new HashSet<int>();
        var res = new int[2];

        for (int i = 0; i < n; i++) {
            if (s.Contains(nums[i])) {
                res[0] = nums[i];
                continue;
            }
            s.Add(nums[i]);
        }

        for (int i = 1; i <= n; i++) {
            if (!s.Contains(i)) {
                res[1] = i;
                break;
            }
        }

        return res;
    }
}